src/Conf.ml

```diff
@@ -1721,11 +1727,14 @@ let validate () =
     `Error (false, "Must specify at least one input file, or `-` for stdin")
   else if has_stdin && inputs_len > 1 then
     `Error (false, "Cannot specify stdin together with other inputs")
-  else if has_stdin && Option.is_none !kind then
+  else if
+    has_stdin && Option.is_none !kind
+    && Option.is_none (Option.bind ~f:kind_of_ext !name)
+  then
     `Error
       ( false
-      , "Must specify at least one of --impl, --intf or --use-file when \
-         reading from stdin" )
+      , "Must specify at least one of --name, --impl or --intf when reading \
```

Comment:
I removed `--use-file` from the message because it's deprecated now and is equivalent to `--impl`